src/ImageSharp/MetaData/ImageMetaData.cs

```diff
@@ -0,0 +1,169 @@
+﻿// <copyright file="ImageMetaData.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+
+    /// <summary>
+    /// Encapsulates the metadata of an image.
+    /// </summary>
+    public sealed class ImageMetaData : IMetaData
+    {
+        /// <summary>
+        /// The default horizontal resolution value (dots per inch) in x direction.
+        /// <remarks>The default value is 96 dots per inch.</remarks>
+        /// </summary>
+        public const double DefaultHorizontalResolution = 96;
+
+        /// <summary>
+        /// The default vertical resolution value (dots per inch) in y direction.
+        /// <remarks>The default value is 96 dots per inch.</remarks>
+        /// </summary>
+        public const double DefaultVerticalResolution = 96;
+
+        private double horizontalResolution;
+        private double verticalResolution;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageMetaData"/> class.
+        /// </summary>
+        internal ImageMetaData()
+        {
+            this.horizontalResolution = DefaultHorizontalResolution;
+            this.verticalResolution = DefaultVerticalResolution;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageMetaData"/> class
+        /// by making a copy from other metadata.
+        /// </summary>
+        /// <param name="other">
+        /// The other <see cref="ImageMetaData"/> to create this instance from.
+        /// </param>
+        internal ImageMetaData(ImageMetaData other)
+        {
+            Debug.Assert(other != null);
+
+            this.HorizontalResolution = other.HorizontalResolution;
+            this.VerticalResolution = other.VerticalResolution;
+            this.Quality = other.Quality;
+            this.FrameDelay = other.FrameDelay;
+            this.RepeatCount = other.RepeatCount;
+
+            foreach (ImageProperty property in other.Properties)
+            {
+              this.Properties.Add(new ImageProperty(property));
+            }
+
+            if (other.ExifProfile != null)
+            {
+              this.ExifProfile = new ExifProfile(other.ExifProfile);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the resolution of the image in x- direction. It is defined as
+        ///  number of dots per inch and should be an positive value.
+        /// </summary>
+        /// <value>The density of the image in x- direction.</value>
+        public double HorizontalResolution
+        {
+            get
+            {
+                return this.horizontalResolution;
+            }
+
+            set
+            {
+              if (value > 0)
```

Comment:
I wonder why do we silently ignore `<=0` values instead of throwing an exception. Is this some kind of standard imaging behavior I don't know about? :)